packages/server/src/app/services/auth/user.service.ts

```diff
@@ -0,0 +1,105 @@
+import {
+    Injectable,
+    Logger,
+    UnauthorizedException,
+    NotFoundException,
+    InternalServerErrorException,
+} from '@nestjs/common';
+import { verify, argon2id } from 'argon2';
+
+import { UsersStore } from '../../db/users/users.store';
+import { JwtPayload } from '@dragonfish/models/auth';
+import {
+    FrontendUser,
+    ChangeUsername,
+    ChangeEmail,
+    ChangePassword,
+    ChangeProfile,
+    UpdateTagline,
+} from '@dragonfish/models/users';
+import { IUser } from '../../shared/auth';
+
+@Injectable()
+export class UserService implements IUser {
+    private readonly logger: Logger = new Logger(UserService.name);
+
+    constructor(private readonly usersService: UsersStore) {}
+
+    async getOneUser(userId: string): Promise<FrontendUser> {
+        return await this.usersService.getOneUser(userId);
+    }
+
+    async changeUsername(jwtPayload: JwtPayload, changeUsernameRequest: ChangeUsername): Promise<FrontendUser> {
+        const potentialUser = await this.usersService.findOneById(jwtPayload.sub);
+        if (!potentialUser) {
+            // This happening is _super_ fishy. Either a well-meaning tinkerer playing with the API, or something malicious.
```

Comment:
Like this, for example: this is a great place to log a warning _AND_ throw, because this is a _weird_ and potentially alarming thing.